src/EditorFeatures/Core/Shared/Preview/PreviewSolutionCrawlerRegistrationService.cs

```diff
@@ -68,6 +74,34 @@ public void Unregister(Workspace workspace, bool blockingShutdown = false)
             {
                 source.Cancel();
             }
+
+            ClearDiagnostics(workspace);
+
+            AssertCleanup();
+        }
+
+        private void ClearDiagnostics(Workspace workspace)
+        {
+            var analyzer = _provider.CreateIncrementalAnalyzer(workspace);
+            var args = _service.GetDiagnosticsUpdatedEventArgs(workspace, projectId: null, documentId: null, cancellationToken: CancellationToken.None);
```

Comment:
sure I can clean that code a bit except RemoveDocument. I can't use ResetAsync since I need synchronous one for clean up.

basically removing a document will clear up all diagnostic associated with that document.

what's how VS workspace clean up when project/solution is removed as well.

project one is not called since we never asked project one to be analyzed.